### DISCUSSION TITLE
`mainAxis` vs `crossAxis`

### DISCUSSION THREAD

**Discussion by KutnerUri** [floating-ui/floating-ui#2757] [General]
Popper uses the terms `mainAxis` and `crossAxis` when talking about the line between the anchor and the popper, and the perpendicular line.

This is however not consistent:

> Flip:
> ```
> y-axis for 'top' and 'bottom' placements
> x-axis for 'left' and 'right' placements
> ```
> 

> offset:
> ```
> default: 0
> 
> The axis that runs along the side of the floating element. Represents the distance (gutter or margin) between the floating element and the reference element.
> ```
> 

Shift is reversed, because its default behavior applies for the secondary axis:
> shift
> ```
> This is the main axis in which shifting is applied.
> 
> x-axis for 'top' and 'bottom' placements
> y-axis for 'left' and 'right' placements
> ```
> 

Obviously changing shift would be a breaking change,
so, I would add a comment in the docs that the axis is opposite to "flip" and "offset", and use the consistent term for future middlewares. What do you think?

**Comment by atomiks** [comment]
This is something I've thought about and the terms "main" and "cross" relate to the main or most useful axis of the middleware's behavior, which is context dependent.  

`sideAxis` and `alignmentAxis` (already used in `offset`) are likely better terms which would remain consistent across each of them. I can convert this to an issue and label it as a Breaking Change for the next major version.

**Reply by KutnerUri** [reply]
idk, `main` and `cross` are very clear.
I understand 'side' is consistent with `placement`, reffering to polygon side, but side still sounds like "secondly" (like "side hustle")

how about `popAxis`(`floatAxis`?) and `alignAxis`?
also we can discuss it a bit before making a PR

**Reply by atomiks** [reply]
I don't think `pop` or `float` convey meaning by themselves whereas `side` refers to something intuitively physical

There's already precedent for the term `Side` - it's referred to throughout the docs as the relevant edge of the reference element to place the floating element (type Placement = `${Side}-${Alignment}`). `useTransitionStyles` also passes in a `side` param. 

While `side` does have that other meaning, I don't think it'll be that ambiguous when paired with `alignmentAxis`, especially if there's a visual.

**Reply by KutnerUri** [reply]
I'm a little cautious, but maybe it will be intuitive if used everywhere.
Not sure if it's worth breaking change, but a docs comment is a good start ("This is the opposite of flip() and offset()")
